### PR TITLE
Implement various score functions

### PR DESCRIPTION
**Pull request by RaffaelBild**

### CODE REVIEW

File: src/main/org/deidentifier/arx/metric/v2/MetricMDNUEntropyPrecomputed.java [hunk 1]
```diff
@@ -333,12 +341,34 @@ protected void initializeInternal(final DataManager manager,
...
         }
         
+        if (config.isPrivacyModelSpecified(DataDependentEDDifferentialPrivacy.class)) {
+            
+            // Store minimal size of equivalence classes
+            DataDependentEDDifferentialPrivacy dpCriterion = config.getPrivacyModel(DataDependentEDDifferentialPrivacy.class);
+            k = (double)dpCriterion.getK();
+            
+            // Store root values of generalization hierarchies or null if no single root value exists
+            rootValues = new Integer[hierarchies.length];
```
Comment (prasser) [inline src/main/org/deidentifier/arx/metric/v2/MetricMDNUEntropyPrecomputed.java]: Why is this array of type Integer[]. int[] is sufficient. You can encode a missing root value as -1, for example.

File: src/main/org/deidentifier/arx/metric/v2/MetricMDNUEntropyPrecomputed.java [hunk 2]
```diff
@@ -326,7 +341,73 @@ protected void initializeInternal(final DataManager manager,
             max[i] = (input.getDataLength() * log2(input.getDataLength())) * Math.max(gFactor, sFactor);
         }
         
+        if (config.isPrivacyModelSpecified(DataDependentEDDifferentialPrivacy.class)) {
+            
+            // Store minimal size of equivalence classes
+            DataDependentEDDifferentialPrivacy dpCriterion = config.getPrivacyModel(DataDependentEDDifferentialPrivacy.class);
+            k = (double)dpCriterion.getK();
```
Comment (prasser) [inline src/main/org/deidentifier/arx/metric/v2/MetricMDNUEntropyPrecomputed.java]: The minimal class size is extracted at multiple locations, so this comment should be considered wherever applicable:

Generic access to minimal class-size

        // For each
        for (PrivacyCriterion c : this.getPrivacyModels()) {
            if (c.isMinimalClassSizeAvailable()) {
                result = Math.max(result, c.getMinimalClassSize());
            }
        }

In this case: dpCriterion.getMinimalClassSize();

Comment (prasser) [reply src/main/org/deidentifier/arx/metric/v2/MetricMDNUEntropyPrecomputed.java]: In the same context: please also overwrite getRiskThresholdJournalist(), getRiskThresholdMarkter() and getRiskThresholdProsecutor() in AbstractDPCriterion and return 1/k in each case.

Comment (prasser) [reply src/main/org/deidentifier/arx/metric/v2/MetricMDNUEntropyPrecomputed.java]: Unrelated issue: can we rename the following classes please (if this does not introduce serializability issues):
AbstractEDDifferentialPrivacy -> EDDifferentialPrivacyAbstract
DataDependentEDDifferentialPrivacy -> EDDifferentialPrivacyOptimal

**Comment by RaffaelBild** [comment]
Thanks you for the feedback! I have addressed your comments.

**Pull request closed without merging**